### PR TITLE
sci-libs/dsdp: revbump (EAPI 4 -> 6)

### PR DESCRIPTION
**Pull request by g-braeunlich**
Package-Manager: portage-2.2.28

### CODE REVIEW

File: sci-libs/dsdp/dsdp-5.8-r3.ebuild [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+	emake OPTFLAGS="${CFLAGS} -fPIC" dsdplibrary
+	make_shared_lib lib/lib${PN}.a || die "doing shared lib failed"
+}
+
+src_test() {
+	emake -j1 example test
+}
+
+src_install() {
+	dolib.so lib/lib${PN}$(get_libname $(get_major_version))
```
Comment (SoapGentoo) [inline sci-libs/dsdp/dsdp-5.8-r3.ebuild]: `get_libname` is part of `multilib`, so you will need to bring back `multilib.eclass`

Comment (SoapGentoo) [inline sci-libs/dsdp/dsdp-5.8-r3.ebuild]: `get_major_version` is part of `versionator`

File: sci-libs/dsdp/dsdp-5.8-r3.ebuild [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+	dosym lib${PN}$(get_libname $(get_major_version)) \
+		/usr/$(get_libdir)/lib${PN}$(get_libname)
+
+	doheader include/*.h src/sdp/*.h
+
+	if use doc; then
+		DOCS+=( docs/*.pdf )
+	fi
```
Comment (SoapGentoo) [inline sci-libs/dsdp/dsdp-5.8-r3.ebuild]: make this more succinct: `use doc && DOCS+=( docs/*.pdf )`
and do the same for examples

File: sci-libs/dsdp/dsdp-5.8-r3.ebuild [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+IUSE="doc examples"
+
+RDEPEND="virtual/lapack"
+DEPEND="${RDEPEND}
+	virtual/pkgconfig"
+
+S="${WORKDIR}/${MYP}"
+
+PATCHES=(
+		"${FILESDIR}"/${P}-readsdpa.patch
```
Comment (SoapGentoo) [inline sci-libs/dsdp/dsdp-5.8-r3.ebuild]: too much indentation

File: sci-libs/dsdp/dsdp-5.8-r3.ebuild [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+make_shared_lib() {
+	local soname=$(basename "${1%.a}")$(get_libname $(get_major_version))
+	einfo "Making ${soname}"
+	${2:-$(tc-getCC)} ${LDFLAGS}  \
+		-shared -Wl,-soname="${soname}" \
+		$([[ ${CHOST} == *-darwin* ]] && echo "-Wl,-install_name -Wl,${EPREFIX}/usr/$(get_libdir)/${soname}") \
+		-Wl,--whole-archive "${1}" -Wl,--no-whole-archive \
+		-o $(dirname "${1}")/"${soname}" \
+		-lm $($(tc-getPKG_CONFIG) --libs blas lapack) || return 1
+
```
Comment (SoapGentoo) [inline sci-libs/dsdp/dsdp-5.8-r3.ebuild]: superfluous free line

File: sci-libs/dsdp/dsdp-5.8-r3.ebuild [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+		-Wl,--whole-archive "${1}" -Wl,--no-whole-archive \
+		-o $(dirname "${1}")/"${soname}" \
+		-lm $($(tc-getPKG_CONFIG) --libs blas lapack) || return 1
+}
+
+src_prepare() {
+	default
+	# to do proper parallel compilation
+	find . -name Makefile -exec \
+		 sed -i -e 's:make :$(MAKE) :g' '{}' \;
```
Comment (SoapGentoo) [inline sci-libs/dsdp/dsdp-5.8-r3.ebuild]: change this to
```
while IFS="" read -d $'\0' -r file; do
	sed -i -e 's:make :$(MAKE) :g' "${file}" || die
done < <(find . -name Makefile -print0)
```
and check that it performs the same changes.

File: sci-libs/dsdp/dsdp-5.8-r3.ebuild [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+	sed -i -e 's:make clean:$(MAKE) clean:g' make.include
```
Comment (SoapGentoo) [inline sci-libs/dsdp/dsdp-5.8-r3.ebuild]: missing `|| die`

File: sci-libs/dsdp/dsdp-5.8-r3.ebuild (hunk 5 above)
Comment (g-braeunlich) [inline sci-libs/dsdp/dsdp-5.8-r3.ebuild]: Works. After src_repare, old version and new versions of ${TMPDIR} agree (except environment)

**Pull request closed without merging**